pallets/automation-time/src/lib.rs

```diff
@@ -251,5 +292,56 @@ pub mod pallet {
 
 			Ok(())
 		}
+
+		/// Run tasks for the block time.
+		///
+		/// Complete as many tasks as possible given the maximum weight.
+		/// Return the weight that was used.
+		///
+		/// Until the TODO is completed we will be limited to two tasks per slot to ensure
+		/// they can all be completed.
+		///
+		/// TODO (ENG-157):
+		/// - Calculate weights for each task
+		pub fn run_tasks(max_weight: Weight) -> Weight {
+			let mut weight_left: Weight = max_weight - RUN_TASK_OVERHEAD;
+			let time_block = Self::get_current_time_slot();
+
+			if let Some(task_ids) = Self::get_scheduled_tasks(time_block) {
+				let mut consumed_task_index: usize = 0;
+				for task_id in task_ids.iter() {
+					if weight_left < MAX_LOOP_WEIGHT {
```

Comment:
good place to throw an err (whenever we figure that out)

Reply:
well we expect this to happen a lot of the time. The overflow case (which I still need to code) would for sure need the error